### PR TITLE
Channel API documentation

### PR DESCRIPTION
**Pull request by raskchanky**
This is the beginning of the modifications to channels.  I started with the API documentation, figuring that was a good place to make sure the API was the way we wanted it before starting on the implementation.

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/doc/api.raml [hunk 1]
```diff
@@ -252,65 +252,207 @@ securitySchemes:
...
+                    400:
+                        description: Origin or channel not supplied
+                    404:
+                        description: Origin or channel does not exist
+                    500:
+                        description: Server error
+            /rename:
+                /{new_name}:
+                    put:
+                        description: Renames a channel
```
Comment (jmink) [inline components/builder-depot/doc/api.raml]: Is a rename safe?

Comment (raskchanky) [inline components/builder-depot/doc/api.raml]: Safe?

Comment (raskchanky) [inline components/builder-depot/doc/api.raml]: I'll just remove the renaming for now.

**Comment by smurawski** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/196033667).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/196033667).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**